pkg/operation/botanist/component/vpnshoot/vpnshoot.go

```diff
@@ -684,6 +708,41 @@ func (v *vpnShoot) getVolumes(secretCA, secret, secretTLSAuth, secretDH *corev1.
 				},
 			},
 		})
+	} else {
+		volumes = append(volumes, corev1.Volume{
+			Name: volumeNameDevNetTun,
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: volumeMountPathDevNetTun,
+					Type: (*corev1.HostPathType)(pointer.String(string(corev1.HostPathCharDev))),
+				},
+			},
+		})
 	}
 	return volumes
 }
+
+func (v *vpnShoot) getInitContainers() []corev1.Container {
+	if !v.values.ReversedVPN.Enabled {
+		return []corev1.Container{}
+	}
+	return []corev1.Container{
+		{
+			Name:            initContainerName,
+			Image:           v.values.Image,
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			Env:             v.getEnvVars(true),
+			SecurityContext: &corev1.SecurityContext{
+				Privileged: pointer.Bool(true),
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("100m"),
+					corev1.ResourceMemory: resource.MustParse("100Mi"),
+				},
```

Comment:
These are the same requests that are used for the vpn-shoot Pod itslef? Why we give the init container that high requests when it is just (supposed to be) a lightweight container that does 4 sysctls and exits.

Reply:
Yes, the requests are identical to the `vpn-shoot` pod. We could indeed go lower with the requests, but I do not really see the point as the resource requests anyway need to be fulfilled after the init container completes to run the real workload. As you already pointed out, the init container should finish fairly fast because it only executes a few writes into kernel space.
In case you strongly feel that we should reduce the requests please also provide what you have in mind.

Reply:
I would suggest:
```go
			Resources: corev1.ResourceRequirements{
				Requests: corev1.ResourceList{
					corev1.ResourceCPU:    resource.MustParse("30m"),
					corev1.ResourceMemory: resource.MustParse("32Mi"),
				},
				Limits: corev1.ResourceList{
					corev1.ResourceCPU:    resource.MustParse("30m"),
					corev1.ResourceMemory: resource.MustParse("32Mi"),
				},
			},
```

---

I would also vote to modify `containerName: '*'` in the VPA spec to `containerName: '<container-name>'` just to make sure that VPA does not bother the init container (I assume that the VPA does not support init containers but just to be on the safe side).

Reply:
I adapted the resource requests/limits accordingly, but did not specify a cpu limit aligned with our general policy.
In addition to that, I also adapted the vpa resource accordingly.